tmedia 781 ci: Use github actor and token rather than personal pats

JackHowa opened the pull request:
## Ticket

- [TMEDIA-781](https://arcpublishing.atlassian.net/browse/TMEDIA-781)

## Description

Replace pats with github token and github actor

## Acceptance Criteria

- should still publish on github action 

## Test Steps

- Tested on same access and protection in this repo
- Approved by @manchicken in appsec

https://github.com/WPMedia/themes-lerna-playground/blob/alpha/.github/workflows/alpha-publish-push-version.yml
https://github.com/WPMedia/themes-lerna-playground/runs/5839474496?check_suite_focus=true

Pretty cool that the commit still shows as the person who committed https://github.com/WPMedia/themes-lerna-playground/commit/b39f539030add48195485cbfe378714924b7f40a

```
commit b39f539030add48195485cbfe378714924b7f40a (HEAD -> alpha)
Author: JackHowa <JackHowa@users.noreply.github.com>
Date:   Tue Apr 5 19:00:11 2022 +0000

    1.0.2-alpha.1
```


## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed relevant documentation has been updated/added.
- [ ] Add label - **ready for review** when the pull request is ready for someone to begin reviewing

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] All GitHub Actions pass
- [ ] Ran the code locally based on the test instructions.
- [ ] Checked Chromatic for Storybook changes, accepted the updates if acceptable
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.
- [ ] Approve and Add label - **ready to merge** if you are happy with the pull request
- [ ] Want another reviewer? Add the label **additional review**

JackHowa committed:
ci: Use github actor and token rather than personal pats

RobinGiannattasio commented on .github/workflows/rc-publish.yml:
```diff
@@ -33,17 +33,15 @@ jobs:
       - name: Prerelease with tag, eg (`0.0.2-rc.{hash of commit}.{current-date-time}`)
-        run: npm version prerelease --preid=rc.$(git rev-parse --short HEAD).${{ steps.get-datetime.outputs.date }}
+        run: |
+          git config --local user.email "$GITHUB_ACTOR@users.noreply.github.com"
```
Okay, I looked it up - but I would love for you to explain it to me :D

Github Actor: `The username of the user that initiated the workflow run.`

Does this mean whoever triggers the action will be the actor? And that would either manually from the github ui or triggering it through the local?

JackHowa replied:
yes! thanks for asking and discussing this in the meeting

JackHowa replied:
@RobinGiannattasio updated -- hope it's ok with the comment added to clarify your q.

JackHowa replied:
will resolve as I think we covered this

RobinGiannattasio reviewed (approved):
This *looks* good, just had a question where I was hoping you could impart some knowledge :D

JackHowa committed:
docs: Add code comment to clarify usage of github actor

Per comment https://github.com/WPMedia/arc-themes-components/pull/62/files#r845223466
